crates/resource-resolver/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+//!
+//! Resource resolver for apps that was packaged by [`cargo-packager`](https://docs.rs/cargo-packager).
+//!
+//! It resolves the root path which contains resources, which was set using the `resources`
+//! field of [cargo packager configuration](https://docs.rs/cargo-packager/latest/cargo_packager/config/struct.Config.html).
+//!
+//! ## Get the resource path
+//!
+//! ```
+//! use cargo_packager_resource_resolver::{resources_dir, PackageFormat};
+//!
+//! let resource_path = resources_dir(PackageFormat::Nsis).unwrap();
+//! ```
+//!
+
+use error::Result;
+use std::{
+    env,
+    path::{Path, PathBuf},
+};
+
+mod error;
+
+pub use error::Error;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum PackageFormat {
+    /// When no format is used (`cargo run`)
+    None,
+    /// The macOS application bundle (.app).
+    App,
+    /// The macOS DMG package (.dmg).
+    Dmg,
+    /// The Microsoft Software Installer (.msi) through WiX Toolset.
+    Wix,
+    /// The NSIS installer (.exe).
+    Nsis,
+    /// The Linux Debian package (.deb).
+    Deb,
+    /// The Linux AppImage package (.AppImage).
+    AppImage,
+}
+
+/// Get the current package format.
+/// Can only be used if the app was build with cargo-packager
+/// and the `before-each-package-command` atribute.
+#[cfg(feature = "auto-detect-format")]
+#[must_use]
+pub fn current_format() -> PackageFormat {
+    // sync with PackageFormat::short_name function of packager crate
+    // maybe having a special crate for the Config struct,
+    // that both packager and resource-resolver could be a
+    // better alternative
+    if cfg!(CARGO_PACKAGER_FORMAT = "app") {
+        PackageFormat::App
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "dmg") {
+        PackageFormat::Dmg
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "wix") {
+        PackageFormat::Wix
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "nsis") {
+        PackageFormat::Nsis
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "deb") {
+        PackageFormat::Deb
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "appimage") {
+        PackageFormat::AppImage
+    } else {
+        PackageFormat::None
+    }
+}
+
+/// Retreive the resource path of your app, packaged with cargo packager.
+/// This function behave the same as [`resource_dir`], except it accepts
+/// a parameter that will be happened to the resource path when no packaging format
+/// is used.
+///
+/// Example: You want to include the folder `crate/resource/icons/`.
+///
+/// - With `cargo run` command, you will have to execute
+///     `resource_dir().unwrap().join("resource/icons/")` to get the path.
+/// - With any other formats, it will be `resource_dir().unwrap().join("icons/")`.
+///
+/// ```
+/// use cargo_packager_resource_resolver::{resolve_resource, PackageFormat};
+///
+/// resolve_resource(PackageFormat::None, "resource").unwrap().join("icons/");
+/// ```
+pub fn resolve_resource<P: AsRef<Path>>(package_format: PackageFormat, path: P) -> Result<PathBuf> {
```

Comment:
this function name is kinda weird, in Tauri it appends the `path` to the resources path, but here it's more of a fallback for the None package format.. can't we force the user to provide that path via the None variant itself? like `PackageFormat::None { local_resources_path: &'static str }`

Reply:
I think we should just drop the function, because it will require cargo packager to export an env var.

Reply:
done in the last commit